sycl/test-e2e/Graph/Explicit/add_nodes_after_finalize.cpp

```diff
@@ -0,0 +1,97 @@
+// REQUIRES: level_zero, gpu
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+// RUN: %if ext_oneapi_level_zero %{env ZE_DEBUG=4 %{run} %t.out 2>&1 | FileCheck %s %}
```

Comment:
Are you sure this works? I'm looking at

https://github.com/intel/llvm/blob/dfb75675bc0ac0406f3845485c7dffb67fbe21a9/sycl/test-e2e/lit.cfg.py#L421-L424

and thinking that the working spelling would be just `%if level_zero`.

Also, we might need `UNSUPPORTED: ze_debug` here - please try running `llvm-lit --param ze_debug=6` to see if the test can pass without it.

Reply:
> Running with --param ze_debug=6 causes the regular runs of the test to fail with PI_ERROR_DEVICE_NOT_FOUND

Can you try removing `build/lib/libze_loader*`? I remember seeing something that was "fixable" like this.

Reply:
Why do you need `%if` at all if there is a requirement to only use level_zero (via `REQUIRES:`)?

Reply:
> Can you try removing build/lib/libze_loader*? I remember seeing something that was "fixable" like this.

That does indeed fix it. Is this just a local build issue then? 

> Why do you need %if at all if there is a requirement to only use level_zero (via REQUIRES:)?

I suppose it is technically not required right now but is more future proofing, since in the future when we have more backend support we will either remove Level Zero from `REQUIRES: ` or expand it to include other backends. If there is no output then the file check will fail, so I think we have to restrict it to running only when we know it will actually generate output.

Reply:
> That does indeed fix it. Is this just a local build issue then?

I don't know the current state of it nor understand the full picture. Does it reappear during rebuild? If so, probably more staff in the CMakeLists.txt has to be dropped. @smaslov-intel and @jandres742 could know more.

Reply:
> I don't know the current state of it nor understand the full picture. Does it reappear during rebuild? If so, probably more staff in the CMakeLists.txt has to be dropped.
Indeed it does reappear when I build again.